packages/txservice/src/dispatch.ts

```diff
@@ -641,9 +662,11 @@ export class TransactionDispatch extends ChainRpcProvider {
     // Here we wait for the target confirmations.
     // TODO: Ensure we are comfortable with how this timeout period is calculated.
     const timeout = this.confirmationTimeout * this.confirmationsRequired * 2;
-    const confirmResult = await this.confirmTransaction(transaction, this.confirmationsRequired, timeout);
-    if (confirmResult.isErr()) {
-      // No other errors should occur during this confirmation attempt. This could occur during a reorg.
+    let receipt: providers.TransactionReceipt;
+    try {
+      receipt = await this.confirmTransaction(transaction, this.confirmationsRequired, timeout);
+    } catch (error) {
+      // No other errors should normally occur during this confirmation attempt. This could occur during a reorg.
       throw new NotEnoughConfirmations(
         this.confirmationsRequired,
```

Comment:
we should log the error with `jsonifyError` just in case

Reply:
will do